tests/integration/test_interface.py

```diff
@@ -78,53 +111,71 @@ def test_generate_dataset_from_sample_and_source(
+    check_noised_sample, check_original_sample, shared_sample_idx = _get_common_datasets(
+        dataset_name, data, noised_sample
     )
-    shared_idx_dataset = pd.Index(
-        set(check_noised_dataset.index).intersection(set(check_original.index))
+    check_noised_dataset, check_original_dataset, shared_dataset_idx = _get_common_datasets(
+        dataset_name, data, noised_dataset
     )
-    for col in check_noised_dataset.columns:
-        original_missing_idx = check_original.index[check_original[col].isna()]
-        both_missing_sample_idx = check_noised_sample.index[
-            check_noised_sample[col].isna()
-        ].intersection(original_missing_idx)
-        both_missing_dataset_idx = check_noised_dataset.index[
-            check_noised_dataset[col].isna()
-        ].intersection(original_missing_idx)
-        compare_sample_idx = shared_idx_sample.difference(both_missing_sample_idx)
-        compare_dataset_idx = shared_idx_dataset.difference(both_missing_dataset_idx)
-        noise_level_single_dataset = (
-            check_original.loc[compare_sample_idx, col].values
-            != check_noised_sample.loc[compare_sample_idx, col].values
-        ).mean()
-        noise_level_full_dataset = (
-            check_original.loc[compare_dataset_idx, col].values
-            != check_noised_dataset.loc[compare_dataset_idx, col].values
-        ).mean()
-        # If the dataset is very small and/or missingness is very high, we can't
-        # meaningfully compare because the stochastic variation is so great
-        # As of 8/31/2023, this only skips unit_number (very missing) in the
-        # ACS and CPS datasets (very small)
-        if len(compare_sample_idx) < 30 or len(compare_dataset_idx) < 30:
-            warnings.warn(
-                f"Noise levels in {col} of {dataset_name} were not compared because the numbers of rows eligible "
-                f"for noise were only {len(compare_sample_idx)} and {len(compare_dataset_idx)}"
+
+    config = get_configuration(config)
+    for col_name in check_noised_sample.columns:
+        col = COLUMNS.get_column(col_name)
+
+        # Check that originally missing data remained missing
+        originally_missing_sample_idx = check_original_sample.index[
+            check_original_sample[col.name].isna()
+        ]
+        originally_missing_dataset_idx = check_original_dataset.index[
+            check_original_dataset[col.name].isna()
+        ]
+        assert check_noised_sample.loc[originally_missing_sample_idx, col.name].isna().all()
+        assert check_noised_dataset.loc[originally_missing_dataset_idx, col.name].isna().all()
+
+        # Check for noising where applicable
+        to_compare_sample_idx = shared_sample_idx.difference(originally_missing_sample_idx)
+        to_compare_dataset_idx = shared_dataset_idx.difference(originally_missing_dataset_idx)
+        if col.noise_types:
```

Comment:
I highly recommend creating a helper funciton to do all of this stuff and then just passing the required args for sample and dataset into it.